pkg/registry/apis/provisioning/register.go

```diff
@@ -314,13 +340,14 @@ func (b *ProvisioningAPIBuilder) beginUpdate(ctx context.Context, obj, old runti
 		return nil, fmt.Errorf("get old repository: %w", err)
 	}
 
+	// Should this be an error -- eg, block the update, or change the health status?
 	if err := b.ensureRepositoryFolderExists(ctx, objCfg); err != nil {
 		return nil, fmt.Errorf("failed to ensure the configured folder exists: %w", err)
 	}
 
 	undo, err := repo.BeginUpdate(ctx, b.logger, oldRepo)
 	if err != nil {
-		return nil, err
```

Comment:
failing here means we are unable to save a github repo with bad token -- even if the goal is to mark it as `healthy=false`!